data/huion-kamvas-pro-24.tablet

```diff
@@ -0,0 +1,60 @@
+# Huion
+# Kamvas Pro 24
+# GT-240
+#
+# sysinfo.0LZKOwtz0e.tar.gz
+# https://github.com/linuxwacom/wacom-hid-descriptors/issues/351
+# Button Map:
+# (A=1, B=2, C=3, ...)
+#
+#          *-----------------------*
+#    A     |                       |     K
+#    B     |                       |     L
+#    C     |                       |     M
+#    D     |                       |     N
+#    E     |                       |     O
+#          |        DISPLAY        |
+#    F     |                       |     P
+#    G     |                       |     Q
+#    H     |                       |     R
+#    I     |                       |     S
+#    J     |                       |     T
+#          *-----------------------*
+#
+# Touch Strip Map:
+# (A=1st strip, B=2nd strip, ...)
+#
+#
+#    *-----------------------*
+#    |                       |
+#  A |                       | B
+#  a |        DISPLAY        | b
+#    |                       |
+#    *-----------------------*
+#
+# Also matches:
+# GAOMON S56K which doesn't have buttons but since HUION re-uses usb IDs
+# there is nothing we can do about this.
+
+[Device]
+Name=Huion Kamvas Pro 24
+ModelName=GT-240
+Class=Cintiq
+DeviceMatch=usb:256c:006e:Tablet Monitor Pen;usb:256c:006e:Tablet Monitor Pad;usb:256c:006e:Tablet Monitor Touch Strip;
```

Comment:
```suggestion
DeviceMatch=usb|256c|006e|Tablet Monitor Pen;usb|256c|006e|Tablet Monitor Pad;usb|256c|006e|Tablet Monitor Touch Strip;
```
once rebased because #648 changed from `:` to `|`

Reply:
@whot The [Kamvas Pro 13](https://github.com/linuxwacom/libwacom/blob/master/data/kamvas-pro-13.tablet) is what Linux currently detects the device as. Is this PR going to remain faulty after #648?

Reply:
yes, we neeed to add the firmware match for this device and for that you need @JoseExposito's kernel patches. Then extract the `uniq` value and add them to the match please.

The `:` to `|` needs to be done either way though, might as well get that out of the way now.

Reply:
@JoseExposito is there a way to find this value without the kernel patch? I don't know how to build the Linux kernel

Reply:
For simplicity, you can build DIGImend instead of the kernel as explained here:
https://gitlab.freedesktop.org/libinput/libinput/-/issues/989#note_2371804

If my [tool worked correctly](https://github.com/linuxwacom/libwacom/issues/610#issuecomment-1857837171), the right value should be `HUION_M184`.

Reply:
@jexposit so I should just put in ``HUION_M184``, or do the instructions you provide have a way of confirming the firmware version for myself (if so I must have missed  them)?

Reply:
The easiest way to test it is adding it:

```
DeviceMatch=usb|256c|006e|Tablet Monitor Pen|HUION_M184;usb|256c|006e|Tablet Monitor Pad|HUION_M184;usb|256c|006e|Tablet Monitor Touch Strip|HUION_M184;
```

And checking if your device is detected (after updating libwacom's database as described in the readme) by running `libwacom-list-local-devices` or checking your GNOME/KDE table settings.

Have in mind that you'd need to compile and install the latest libwacom code. I don't know if it is documented, but, to run the latest `libwacom-list-local-devices` **without overriding** your system's libwacom:

```
$ git clone https://github.com/linuxwacom/libwacom.git
$ cd libwacom
$ meson setup builddir --prefix=$(pwd)/builddir/install # It might require installing some dependencies
$ ninja -C builddir
$ ninja -C builddir install
$ LD_LIBRARY_PATH=$(pwd)/builddir/install/lib64:$LD_LIBRARY_PATH \
   ./builddir/install/bin/libwacom-list-local-devices
```

That last command should (hopefully) detect your tablet correctly.

If you want GNOME/KDE settings to recognize your tablet, you'd need to override your distro libwacom with the latest master, which I won't recommend unless you are testing in a VM or in a test PC that you plan to format.

Reply:
@jexposit 
![Screenshot from 2024-04-18 22-03-08](https://github.com/linuxwacom/libwacom/assets/64518882/70bf695d-8028-418e-bcb1-bd1fa8a97683)

Looks like things are correctly detected. I think ideally we'd find someone with a Kamvas Pro 13 to test if we've overridden their tablet file before merging.

Do you suppose it would be a good idea to add the Kamvas Pro 13's firmware version ``HUION_M182`` in its tablet file for this PR?

Reply:
> Looks like things are correctly detected. I think ideally we'd find someone with a Kamvas Pro 13 to test if we've overridden their tablet file before merging.

The [current rules](https://github.com/linuxwacom/libwacom/blob/master/libwacom/libwacom.c#L627-L637) to match devices with the same vendor ID, product ID and bus are:

1. Match the `DeviceMatch` value by name and firmware ("uniq" value)
2. If there is no match, try with name only
3. If there is no match, try without name and firmware

Since the `DeviceMatch` of your .table file contains the firmware, only your tablet will match it. Note that using an older kernel (without uniq/firmware support), your tablet will be matched as a Kamvas Pro 13, like it does now.

Reply:
> Since the ``DeviceMatch`` of your .table file contains the firmware, only your tablet will match it. Note that using an older kernel (without uniq/firmware support), your tablet will be matched as a Kamvas Pro 13, like it does now

Then device matching in this PR is resolved?

Reply:
I think so, I can't resolve this thread, but the DeviceMatch seems correct and keeps backwards compatibility :)

Reply:
Apologies, I made an edit. I'm going to resolve this thread and make another one about the consensus on merging the different listed events.